components/sync_manager/ffi/src/lib.rs

```diff
@@ -0,0 +1,84 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#![allow(unknown_lints)]
+#![warn(rust_2018_idioms)]
+// Let's allow these in the FFI code, since it's usually just a coincidence if
+// the closure is small.
+#![allow(clippy::redundant_closure)]
+
+use ffi_support::{ExternError, HandleError};
+use sync_manager::Result as MgrResult;
+
+#[no_mangle]
+pub extern "C" fn sync_manager_set_places(_places_api_handle: u64, error: &mut ExternError) {
+    ffi_support::call_with_result(error, || -> MgrResult<()> {
+        // #[cfg(feature = "places")]
+        {
+            let api = places_ffi::APIS
+                .get_u64(_places_api_handle, |api| -> Result<_, HandleError> {
+                    Ok(std::sync::Arc::clone(api))
+                })?;
+            sync_manager::set_places(api);
+            Ok(())
+        }
+        // #[cfg(not(feature = "places"))]
+        // {
+        //     log::error!("Sync manager not compiled with places support");
+        //     Err(sync_manager::ErrorKind::UnsupportedFeature("places".to_string()).into())
+        // }
+    })
+}
+
+#[no_mangle]
+pub extern "C" fn sync_manager_set_logins(_logins_handle: u64, error: &mut ExternError) {
```

Comment:
the handle stuff here (and in the `getHandle()` stuff elsewhere) is extremely unfortunate, but I guess you've tried hard to avoid it :(